cpp/lib/src/RegexMatcher.cc

```diff
@@ -172,6 +172,65 @@ std::string ThreadSafeRegexMatcher::replaceAll(const std::string &subject, const
 }
 
 
+static std::string InsertReplacement(const ThreadSafeRegexMatcher::MatchResult &result, const std::string &replacement_pattern) {
```

Comment:
Why don't you reuse the already existing implementation?

Reply:
As discussed `ThreadSafeRegexMatcher` returns a  `MatchResult` which forces a different handling.